Dockerfile

```diff
@@ -0,0 +1,46 @@
+# Use Node.js v16 as the base image to run this Dockerfile
+FROM node:16 AS base
+
+# Set the working directory cursor to /od-app
+WORKDIR /od-app
+
+# Copy the following files and directories to the Docker image to work correctly with yarn install command.
+COPY package.json ./
+COPY .yarnrc.yml ./
+COPY yarn.lock ./
+COPY .yarn/ ./.yarn/
+# NOTE: This is a workaround but we can reduce the size of the Docker image by removing the following line.
+COPY . ./
+
+
+# Add ARG for each environment variable
+ARG REACT_APP_NETWORK_ID
+ARG REACT_APP_NETWORK_URL
+ARG REACT_APP_FALLBACK_SUBGRAPH_URL
+ARG REACT_APP_GEOFENCE_ENABLED
+
+# Set ENV for each environment variable
+ENV REACT_APP_NETWORK_ID=${REACT_APP_NETWORK_ID}
+ENV REACT_APP_NETWORK_URL=${REACT_APP_NETWORK_URL}
+ENV REACT_APP_FALLBACK_SUBGRAPH_URL=${REACT_APP_FALLBACK_SUBGRAPH_URL}
+ENV REACT_APP_GEOFENCE_ENABLED=${REACT_APP_GEOFENCE_ENABLED}
```

Comment:
The purpose of this PR is to allow anyone to run the image locally. We do not want to include our sensitive data in the docker image itself. Rather we want the end-user to provide their own environment variables.

- [ ] Remove setting the env during build, and instead env should be passed to the image at runtime
- [ ] Update the docker run command to run the image while passing the env variables. We don't want to require docker-compose, since that requires downloading the repo. Rather a docker run command can be run on any machine pull the image from the registry directly.
- [ ] Update the docker run command to expose the appropriate ports.
- [ ] Remove the `COPY . ./` which could potentially expose our env secrets (eg the RPC) by including them in the image.

Reply:
@pi0neerpat , based on this review now we can run locally the app via 2 commands:
1. `docker build -t open-dollar-app .`
2. `docker run -e REACT_APP_NETWORK_ID=420 -e REACT_APP_NETWORK_URL=https://arbitrum-one.publicnode.com -p 8080:3000 -d open-dollar-app`

No docker compose is needed on this update and also you can pass the env vars to the image at runtime.